src/v2/guide/composition.md

```diff
@@ -0,0 +1,548 @@
+
+<p class="tip">Note that <code>_</code>-prefixed methods are not considered part of the stable API, so they may be changed or removed without warning. Using render helpers is considered relatively safe, but accompanying <a href="unit-testing.html">unit tests</a> are recommended to ensure that everything still works after upgrading Vue to a new version.</p>
+
+### Global Mixins
+
+You can also apply a mixin globally, but do this with caution!
+
+<p class="tip">Global mixins affect <strong>every</strong> Vue instance and component, including third-party components.</p>
+
+One appropriate use case, however, is to inject processing logic for custom options, such as in the [VueFire plugin](https://github.com/vuejs/vuefire#usage):
+
+``` js
+// Inject handler for a `firebase` custom option
+Vue.mixin({
+  created: function () {
+    var firebaseBindings = this.$options.firebase
+    if (firebaseBindings) {
+      // Add instance properties and listeners
+      // for data from firebase.
+    }
+  }
+})
+```
+
+Check out [VueFire's source](https://github.com/search?utf8=%E2%9C%93&q=%22Vue.mixin%22+repo%3Avuejs%2Fvuefire+path%3A%2Fsrc&type=Code) for a complete example.
+
+
+
+## Custom Merge Strategies
+
+Plugins often define custom options, which use the default strategy of overwriting any existing value. If you're the author a plugin where this is undesirable, you can define a **custom merge strategy** for your options with `Vue.config.optionMergeStrategies`.
+
+For example, if your plugin uses a `foo` option:
+
+``` js
+Vue.config.optionMergeStrategies.foo = function (newValue, baseValue) {
+  // Return the correctly merged value
+}
+```
+
+For most object-based options, you can simply use the same strategy used by `methods`:
+
+``` js
+var strategies = Vue.config.optionMergeStrategies
+strategies.foo = strategies.methods
+```
+
+A more advanced example can be found on [Vuex](https://github.com/vuejs/vuex)'s 1.x merging strategy:
+
+``` js
+const merge = Vue.config.optionMergeStrategies.computed
+Vue.config.optionMergeStrategies.vuex = function (newValue, baseValue) {
+  if (!newValue) return baseValue
+  if (!baseValue) return newValue
+  return {
+    getters: merge(newValue.getters, baseValue.getters),
+    state: merge(newValue.state, baseValue.state),
+    actions: merge(newValue.actions, baseValue.actions)
+  }
+}
+```
+
+
+
+## Wrapper Components
+
+As you've seen, advanced composition with extends and mixins can quickly grow complicated. One simpler alternative is creating **wrapper components**, which provide an interface for another component typically used as its root element.
+
+For example, the `ErrorNote` component we [created earlier with extension](#An-Example-Use-Case) could be refactored to:
+
+``` js
+var ErrorNote = {
+  components: {
+    AppNote: AppNote
+  },
+  template: `
+    <AppNote type="error">
+      <span slot="bold">Error!</span>
+      <slot/>
+    </AppNote>
+  `
+}
+```
```

Comment:
I wonder if we should have one more example for styling with a scoped slot because that's pretty useful for theming. That could also go in the cookbook.

Reply:
I was reluctant to include it, because I prefer other patterns for most use cases, but between you, @Akryum, and others, I've been convinced we need a section for it. 🙂

Reply:
OK, I did some work on this today, but I just couldn't make scoped props fit because unlike all the other strategies on this page, they're not about sharing functionality between components. Instead, they're really more for _communication_ between parent-child components, so the examples I came up with will be in a new page on advanced communication between components instead.